cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -0,0 +1,169 @@
+import static io.ballerina.runtime.util.RuntimeConstants.SYSTEM_PROP_BAL_DEBUG;
+import static java.nio.file.Files.createDirectories;
+import static org.ballerinalang.tool.LauncherUtils.createLauncherException;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.SUPPORTED_PLATFORMS;
+
+/**
+ * This class represents the "ballerina pull" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = PULL_COMMAND,
+        description = "download the module source and binaries from a remote repository")
+public class PullCommand implements BLauncherCmd {
+    private PrintStream errStream;
+    private CentralAPIClient client;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    @CommandLine.Option(names = { "--help", "-h" }, hidden = true)
+    private boolean helpFlag;
+
+    @CommandLine.Option(names = "--debug", hidden = true)
+    private String debugPort;
+
+    public PullCommand() {
+        this.errStream = System.err;
+        this.client = new CentralAPIClient();
+    }
+
+    public PullCommand(PrintStream errStream, CentralAPIClient client) {
+        this.errStream = errStream;
+        this.client = client;
+    }
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(PULL_COMMAND);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+
+        if (argList == null || argList.isEmpty()) {
+            throw LauncherUtils.createUsageExceptionWithHelp("no package given");
+        }
+
+        if (argList.size() > 1) {
+            throw LauncherUtils.createUsageExceptionWithHelp("too many arguments");
+        }
+
+        // Enable remote debugging
+        if (null != debugPort) {
+            System.setProperty(SYSTEM_PROP_BAL_DEBUG, debugPort);
+        }
+
+        String resourceName = argList.get(0);
+        String orgName;
+        String packageName;
+        String version;
+
+        if (!validPackageName(resourceName)) {
+            CommandUtil.printError(errStream, "invalid package name. Provide the package name with the org name ",
+                    "ballerina pull {<org-name>/<package-name> | <org-name>/<package-name>:<version>}", false);
+            Runtime.getRuntime().exit(1);
+            return;
+        }
+
+        // Get org name
+        String[] moduleInfo = resourceName.split("/");
+        orgName = moduleInfo[0];
+        String moduleNameAndVersion = moduleInfo[1];
+
+        // Get package name
+        String[] packageInfo = moduleNameAndVersion.split(":");
+        if (packageInfo.length == 2) {
+            packageName = packageInfo[0];
+            version = packageInfo[1];
+        } else {
+            packageName = moduleNameAndVersion;
+            version = Names.EMPTY.getValue();
+        }
+
+        Path packagePathInBaloCache = RepoUtils.createAndGetHomeReposPath()
```

Comment:
Shall we move `RepoUtils.createAndGetHomeReposPath()` to ProjectUtils?

Reply:
Done